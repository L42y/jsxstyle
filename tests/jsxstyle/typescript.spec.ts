import path = require('path');
import * as ts from 'typescript';

// tslint:disable-next-line no-var-requires
const tsConfig = require('jsxstyle/src/tsconfig.json');

const demoFile = path.resolve(__dirname, './typescript/demo.tsx');

it('compiles', () => {
  // TODO: why doesn't TS find `jsx` in compilerOptions?
  const options = { ...tsConfig, jsx: 'react' };
  const host = ts.createCompilerHost(options);
  const program = ts.createProgram([demoFile], options, host);
  const diagnostics = ts.getPreEmitDiagnostics(program);
  let report = '';
  if (diagnostics.length > 0) {
    report = ts.formatDiagnostics(diagnostics, host);
  }
  expect(report).toEqual('');
});
