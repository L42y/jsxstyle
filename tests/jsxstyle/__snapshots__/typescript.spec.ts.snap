// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws type errors for invalid component/prop types 1`] = `
"jsxstyle/typescript/demo.tsx(14,51): error TS2322: Type '{ value: string; typeError: boolean; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> | DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>'.
  Object literal may only specify known properties, and 'typeError' does not exist in type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> | DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>'.
jsxstyle/typescript/demo.tsx(20,39): error TS2322: Type 'true' is not assignable to type 'never'.
jsxstyle/typescript/demo.tsx(24,50): error TS2322: Type 'true' is not assignable to type 'never'.
"
`;
